### PR TITLE
fix: [Model] update does not work when using custom entity with toRawArray()

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
See https://github.com/codeigniter4/CodeIgniter4/pull/8260#discussion_r1408795786

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Model.php [hunk 1]
```diff
@@ -814,12 +818,11 @@ protected function objectToRawArray($data, bool $onlyChanged = true, bool $recur
 
         // Always grab the primary key otherwise updates will fail.
         if (
-            // @TODO Should use `$data instanceof Entity`.
+            // @TODO Should add an interface for this?
             method_exists($data, 'toRawArray')
             && (
                 ! empty($properties)
                 && ! empty($this->primaryKey)
-                && ! in_array($this->primaryKey, $properties, true)
```
Comment (michalsn) [inline system/Model.php:822]: Why are we removing this check?

**Review by michalsn — COMMENTED** [review]
I have a general issue with the use of `$onlyChanged = true`. I know there were good reasons behind it, but the reality is it brings almost zero benefits and is a source of frustration. Hopefully, we can step away from using it someday. But this is just a side note.

Back to the problem... I believe this is okay.

File: system/Model.php (hunk 1 above)
Comment (kenjis) [reply system/Model.php:822]: This does not make sense.
`$this->primaryKey` is the column name like `id`.
`$properties` is an array with `property_name => value`.
If `id` is in the array values, the condition is `false`.

I thought it was a mistake, and it should use `array_key_exists()`, but
when the primary key value is binary data and it is cast to a string by Entity casting,
we need to set raw binary data even when there is a string primary key value in `$properties`.

Without casting, if there is a primary key value in `$properties`, we don't need to set it again,
but setting the same value is no harm.

**Comment by kenjis** [comment]
This bug and the code around it is for `$onlyChanged = true`.
It would not exist without `$onlyChanged`.
Performance would be better with large data and fewer fields to change, but the code is considerably more complex because of `$onlyChanged`.

If we were to create a custom Entity, implementing `$onlyChanged` would be quite tedious. It requires a process to determine the initial values and cannot be easily integrated with the current Model.

File: system/Model.php (hunk 1 above)
Comment (michalsn) [reply system/Model.php:822]: Oh.. yes, that makes sense. Thanks.

**Review by michalsn — APPROVED** [review]

**Comment by kenjis** [comment]
It seems the bug #7797 is not fixed, and something wrong in Model.
I'm checking now.

**Pull request closed without merging**